File: app/src/instrumentTest/java/com/shen/xi/android/tut/test/GuiceInjectorConfigurationTest.java
package com.shen.xi.android.tut.test;

import android.app.Application;

import com.google.api.client.http.HttpRequestFactory;
import com.google.inject.Guice;
import com.google.inject.Inject;
import com.google.inject.Injector;
import com.google.inject.Key;
import com.google.inject.name.Named;
import com.google.inject.name.Names;
import com.shen.xi.android.tut.TuTModule;
import com.shen.xi.android.tut.weibo.WeiboClient;

import junit.framework.TestCase;

import java.util.Collection;


public class GuiceInjectorConfigurationTest extends TestCase {
    private Injector injector;

    public void setUp() throws Exception {
        injector = Guice.createInjector(new TuTModule(new Application()));
    }

    public void testInjectWeiboClient() {
        WeiboClient client = injector.getInstance(WeiboClient.class);

        assertNotNull(client);
        assertTrue(WeiboClient.class.isInstance(client));
    }

    public void testInjectWeiboScope() {
        WeiboScope object = injector.getInstance(WeiboScope.class);

        assertNotNull(object);
        assertFalse(object.getScope().isEmpty());
    }

    public void testInjectQingRequestFactory() {
        HttpRequestFactory instance = injector.getInstance(Key.get(HttpRequestFactory.class, Names.named("qing request factory")));
        assertNotNull(instance);
    }

    private static class WeiboScope {
        private final Collection<String> mScope;

        @Inject
        public WeiboScope(@Named("scope") Collection<String> scope) {

            this.mScope = scope;
        }

        public Collection<String> getScope() {
            return mScope;
        }
    }
}

File: app/src/androidTest/scala/com/shen/xi/android/tut/test/GuiceInjectorConfigurationTest.scala
package com.shen.xi.android.tut.test

import java.util

import android.app.Application
import com.google.api.client.http.HttpRequestFactory
import com.google.inject.name.{Named, Names}
import com.google.inject.{Guice, Inject, Injector, Key}
import com.shen.xi.android.tut.TuTModule
import com.shen.xi.android.tut.weibo.WeiboClient
import junit.framework.TestCase


object GuiceInjectorConfigurationTest {

  private class WeiboScope @Inject()(@Named("scope") scope: util.Collection[String]) {

    def getScope = scope

  }

}

class GuiceInjectorConfigurationTest extends TestCase {

  import com.shen.xi.android.tut.test.GuiceInjectorConfigurationTest.WeiboScope
  import junit.framework.Assert._


  private var injector: Injector = null

  @throws[Exception]
  override def setUp() = injector = Guice.createInjector(new TuTModule(new Application()))

  def testInjectWeiboClient() = {
    val client = injector.getInstance(classOf[WeiboClient])

    assertNotNull(client)
    assertTrue(client.isInstanceOf[WeiboClient])
  }

  def testInjectWeiboScope() = {
    val o = injector.getInstance(classOf[WeiboScope])

    assertNotNull(o)
    assertFalse(o.getScope.isEmpty)
  }

  def testInjectQingRequestFactory() = {
    val instance = injector.getInstance(Key.get(classOf[HttpRequestFactory], Names.named("qing request factory")))

    assertNotNull(instance)
  }

}
